synapseclient/multipart_upload.py

```diff
@@ -350,7 +352,7 @@ def _multipart_upload(syn, filename, contentType, get_chunk_function, md5, fileS
     syn.logger.debug("previously completed %d parts, estimated %d bytes" % (completedParts, previously_completed_bytes))
     time_upload_started = time.time()
     retries = 0
-    mp = Pool(8)
+    mp = PoolProvider(single_threaded=synapseclient.config.single_threaded)
```

Comment:
the pool_provider class should be able to check the config. multipart_upload doesn't need that responsibility